packages/core/src/integer.ts

```diff
@@ -867,7 +869,13 @@ class Integer {
     let result = Integer.ZERO
     for (let i = 0; i < str.length; i += 8) {
       const size = Math.min(8, str.length - i)
-      const value = parseInt(str.substring(i, i + size), radix)
+      const valueString = str.substring(i, i + size)
+      const value = parseInt(valueString, radix)
+
+      if (strictStringValidation === true && !_isValidNumberFromString(valueString, value, radix)) {
+        throw newError(`number format error: "${valueString}" is NaN in radix ${radix}: ${str}`)
```

Comment:
Would this raise a generic error? Perhaps there's some validation error code that could be used? (just interested on change needed)

Reply:
I've kept the error similar the ones we have with other validation errors. It will raise a `Neo4jError` with `code` equals to `N/A`. 

See: https://github.com/neo4j/neo4j-javascript-driver/pull/985/files#diff-654bf635043151007a5b9eecd461ba5d922f5c79bc3e5bc702a75d4d855440eeR843